### PR TITLE
Add environment variable to set container image repository

### PR DESCRIPTION
**Pull request by nhudson**
* Add new environment variable `STACK_IMAGE_REPOSITORY` to allow for alternate container image repositories to be set (eg AWS ECR).
* Move Stack configuration to separate `repository` and `image` configurations.
   * This adds a new field `repository` into the stack configuration [here](https://github.com/tembo-io/tembo/blob/e74dca1273adfcc1a8abd65e360b2179681e1287/tembo-operator/src/stacks/types.rs#L72-L75)
   * This will require a change to the control-plane webserver https://github.com/tembo-io/control-plane/pull/501
   * `{{repository}}` will be replaced with the default value of `quay.io/tembo` or whatever repository is supplied with the `STACK_IMAGE_REPOSITORY` environment variable.
        ```
        name: Standard
        description: A balanced Postgres instance optimized for OLTP workloads.
        repository: "{{repository}}"
        image: "standard-cnpg:15.3.0-1-839d08e"
        stack_version: 0.1.0
        ```
* Bump Helm chart due to CRD regeneration.
* A few code fixes and clippy warning cleanup changes.

### CODE REVIEW

File: tembo-operator/src/extensions/toggle.rs [hunk 1]
```diff
@@ -405,7 +405,7 @@ async fn check_for_extensions_enabled_with_load(
         for extension in trunk_installed_extensions_not_in_all_actually_installed_extensions.clone()
         {
             let found =
-                check_for_so_files(cdb, ctx.clone(), &*pod_name, extension.name.clone()).await?;
+                check_for_so_files(cdb, ctx.clone(), &pod_name, extension.name.clone()).await?;
```
Comment (nhudson) [inline tembo-operator/src/extensions/toggle.rs:408]: This is a clippy warning fix

File: tembo-operator/src/extensions/toggle.rs [hunk 2]
```diff
@@ -430,7 +430,7 @@ async fn check_for_extensions_enabled_with_load(
                         found = true;
                         for desired_location in desired_extension.locations {
                             let location_status = ExtensionInstallLocationStatus {
-                                enabled: Some(desired_location.enabled.clone()),
+                                enabled: Some(desired_location.enabled),
```
Comment (nhudson) [inline tembo-operator/src/extensions/toggle.rs:433]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 3]
```diff
@@ -411,18 +412,12 @@ pub async fn get_loadable_library_name(
...
-            }
-        }
-        None => None,
-    };
+    let loadable_library_name = extension_metadata
+        .loadable_libraries
+        .as_ref()
+        .and_then(|loadable_libraries| loadable_libraries.iter().find(|l| l.requires_restart))
+        .map(|loadable_library| loadable_library.library_name.clone());
+
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:420]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 4]
```diff
@@ -496,8 +491,8 @@ pub fn convert_to_semver(version: String) -> String {
 fn sort_semver(versions: Vec<String>) -> Vec<String> {
     let mut versions = versions;
     versions.sort_by(|a, b| {
-        let a = semver::Version::parse(&a).unwrap();
-        let b = semver::Version::parse(&b).unwrap();
+        let a = semver::Version::parse(a).unwrap();
+        let b = semver::Version::parse(b).unwrap();
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:495]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 5]
```diff
@@ -550,17 +545,17 @@ mod tests {
         let extension_name = "auto_explain".to_string();
         let result = extension_name_matches_trunk_project(extension_name).await;
         assert!(result.is_ok());
-        assert_eq!(result.unwrap(), true);
+        assert!(result.unwrap());
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:548]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 6]
```diff
@@ -550,17 +545,17 @@ mod tests {
...
 
         let extension_name = "pgml".to_string();
         let result = extension_name_matches_trunk_project(extension_name).await;
         assert!(result.is_ok());
-        assert_eq!(result.unwrap(), false);
+        assert!(!result.unwrap());
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:553]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 7]
```diff
@@ -550,17 +545,17 @@ mod tests {
...
 
         let extension_name = "vector".to_string();
         let result = extension_name_matches_trunk_project(extension_name).await;
         assert!(result.is_ok());
-        assert_eq!(result.unwrap(), false);
+        assert!(!result.unwrap());
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:558]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 8]
```diff
@@ -592,7 +587,7 @@ mod tests {
         let version = "15.3.0".to_string();
         let result = is_control_file_absent(trunk_project, version).await;
         assert!(result.is_ok());
-        assert_eq!(result.unwrap(), true);
+        assert!(result.unwrap());
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:590]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 9]
```diff
@@ -618,11 +613,11 @@ mod tests {
     fn test_is_semver() {
         let version = "1.2.3".to_string();
         let result = is_semver(version);
-        assert_eq!(result, true);
+        assert!(result);
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:616]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 10]
```diff
@@ -618,11 +613,11 @@ mod tests {
...
 
         let version = "1.2".to_string();
         let result = is_semver(version);
-        assert_eq!(result, false);
+        assert!(!result);
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:620]: This is a clippy warning fix

File: tembo-operator/tests/integration_tests.rs [hunk 11]
```diff
@@ -613,19 +612,17 @@ mod test {
...
-                        } else {
-                            println!(
-                                "Attempt {}/{}: CoreDB {} has extension {} disabled in status",
-                                attempt, max_retries, name, extension
-                            );
-                        }
+                        println!(
+                            "Attempt {}/{}: CoreDB {} has extension {} disabled in status",
+                            attempt, max_retries, name, extension
+                        );
```
Comment (nhudson) [inline tembo-operator/tests/integration_tests.rs:625]: This is a clippy warning fix

File: tembo-operator/tests/integration_tests.rs [hunk 12]
```diff
@@ -2332,7 +2329,7 @@ mod test {
         });
         let params = PatchParams::apply("tembo-integration-test");
         let patch = Patch::Apply(&coredb_json);
-        let coredb_resource = coredbs.patch(name, &params, &patch).await.unwrap();
+        let _coredb_resource = coredbs.patch(name, &params, &patch).await.unwrap();
```
Comment (nhudson) [inline tembo-operator/tests/integration_tests.rs:2332]: This is a clippy warning fix

File: tembo-operator/tests/integration_tests.rs [hunk 13]
```diff
@@ -2346,7 +2343,7 @@ mod test {
         // Check status.extensions.locations is not empty
         let coredb_resource = coredbs.get(name).await.unwrap();
         for extension in coredb_resource.status.unwrap().extensions.unwrap() {
-            assert!(extension.locations.len() > 0);
+            assert!(!extension.locations.is_empty());
```
Comment (nhudson) [inline tembo-operator/tests/integration_tests.rs:2346]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 14]
```diff
@@ -367,7 +368,7 @@ pub async fn is_control_file_absent(
     //  where there are multiple extensions
     let control_file_absent = project_metadata
         .extensions
-        .get(0)
+        .first()
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:371]: This is a clippy warning fix

File: tembo-operator/src/trunk.rs [hunk 15]
```diff
@@ -277,7 +277,7 @@ pub async fn get_trunk_project_metadata_for_version(
         let response_body = response.text().await?;
         let project_metadata: Vec<TrunkProjectMetadata> = serde_json::from_str(&response_body)?;
         // There will only be one index here, so we can safely assume index 0
-        let project_metadata = match project_metadata.get(0) {
+        let project_metadata = match project_metadata.first() {
```
Comment (nhudson) [inline tembo-operator/src/trunk.rs:280]: This is a clippy warning fix

**Review by sjmiller609 — COMMENTED** [review]
I think we can simplify like this:

- Don't introduce the concept of templating to stack definitions
- No new environment variable

On control plane, we can change the registry to our ECR like this

```
let repository = "hardcoded";
let container_uri = format!("{:?}/{:?}", repository, stack.clone().image);
```

**Pull request closed without merging**